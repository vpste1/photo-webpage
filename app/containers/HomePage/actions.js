/*
 *
 * HomePage actions
 *
 */

import {
  DEFAULT_ACTION,
  PAGE_LOAD,
  URL_SUCCESS,
  URL_FAIL,
} from './constants';

export function defaultAction() {
  return {
    type: DEFAULT_ACTION,
  };
}

export function pageLoadAction() {
  console.log('pageLoadAction');
  return {
    type: PAGE_LOAD,
  };
}

export function retrieveURLsSuccess(titles, imageUrls) {
  console.log('retrieveURLsSuccess');
  console.log(imageUrls);
  return {
    type: URL_SUCCESS,
    payload: { titles,
      imageUrls,
    },
  };
}

export function retrieveURLsFailure() {
  console.log('retrieveURLsFailure');
  return {
    type: URL_FAIL,
  };
}
