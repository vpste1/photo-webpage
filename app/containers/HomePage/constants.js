/*
 *
 * HomePage constants
 *
 */

export const DEFAULT_ACTION = 'app/HomePage/DEFAULT_ACTION';

export const PAGE_LOAD = 'app/HomePage/PAGE_LOAD';

export const URL_SUCCESS = 'app/HomePage/URL_SUCCESS';

export const URL_FAIL = 'app/HomePage/URL_FAIL';
