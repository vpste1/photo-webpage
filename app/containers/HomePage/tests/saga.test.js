import { takeLatest, put } from 'redux-saga/effects';

import watchPageLoad, { pageLoaded } from '../saga';
import { PAGE_LOAD } from '../constants';
import { retrieveURLsSuccess, retrieveURLsFailure } from '../actions';

describe('Container [App] - sagas', () => { /* eslint-disable redux-saga/yield-effects */
  describe('pageLoaded Saga', () => {
    let pageLoadedGenerator;
    beforeEach(() => {
      pageLoadedGenerator = pageLoaded();
      const tokenCallDescriptor = pageLoadedGenerator.next().value;
      console.log('&&&');
      console.log(tokenCallDescriptor);
      expect(tokenCallDescriptor).toMatchSnapshot();

      const childrenCallDescriptor = pageLoadedGenerator.next().value;
      console.log('***&&&');
      console.log(childrenCallDescriptor);
      expect(childrenCallDescriptor).toMatchSnapshot();
    });
    it('should dispatch retrieveURLsSuccess action if the API is hit successfully', () => {
      const titles = ['title1', 'title2'];
      const imageUrls = ['url1', 'url2'];
      const putDescriptor = pageLoadedGenerator.next(titles, imageUrls).value;
      expect(putDescriptor).toEqual(put(retrieveURLsSuccess(titles, imageUrls)));
    });
    it('should dispatch retrieveURLsFailure action if an error is thrown', () => {
      const response = new Error('Unable to reach API');
      const putDescriptor = pageLoadedGenerator.throw(response).value;
      expect(putDescriptor).toEqual(put(retrieveURLsFailure(response)));
    });
  });

  describe('watchPageLoad Saga', () => {
    const watchPageLoadSaga = watchPageLoad();
    it('works with unit tests', () => {
      const takeLatestDescriptor = watchPageLoadSaga.next().value;
      expect(takeLatestDescriptor).toEqual(takeLatest(PAGE_LOAD, pageLoaded));
    });
  });
});
