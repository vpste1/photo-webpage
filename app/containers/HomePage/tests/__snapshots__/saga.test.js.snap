// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container [App] - sagas pageLoaded Saga should dispatch retrieveURLsFailure action if an error is thrown 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Container [App] - sagas pageLoaded Saga should dispatch retrieveURLsFailure action if an error is thrown 2`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "type": "app/HomePage/URL_FAIL",
    },
    "channel": null,
  },
}
`;

exports[`Container [App] - sagas pageLoaded Saga should dispatch retrieveURLsSuccess action if the API is hit successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Container [App] - sagas pageLoaded Saga should dispatch retrieveURLsSuccess action if the API is hit successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "type": "app/HomePage/URL_FAIL",
    },
    "channel": null,
  },
}
`;
